README.md

```diff
@@ -51,13 +51,69 @@ Stop watching file(s).
 - path(String|Array): file path(s)
 - listener(Function): file change callback
 
-## watch mode
-
-[development mode](https://github.com/eggjs/egg-watcher/blob/master/lib/event-sources/development.js) is included.
-
-### development Mode
-
-When your app is running at local(env is `local`), it will be in `development mode`. Once local file is modified it will emit `change` event immediately.
+## Watching mode
+
+### `development` Mode
+
+There's a built-in [development mode](https://github.com/eggjs/egg-watcher/blob/master/lib/event-sources/development.js) which works in local(env is `local`). Once files on disk is modified it will emit a `change` event immediately.
+
+### Customize Watching Mode
+
+Firstly define your custom event source like this:
+
+```js
+// custom_event_source.js
+const Base = require('sdk-base');
+class CustomEventSource extends Base {
+  constructor(opts) {
+    super(opts);
+    this.ready(true);
+  }
+
+  watch(path) {
+    // replace this with your desired way of watching,
+    // when aware of any change, emit a `change` event
+    this._h = setInterval(() => {
+      this.emit('change', {
```

Comment:
yes, but if offer a `type` to `info` object will be nice to developer.

what I mean is should we define the suggest properties of `info` object, so when developer write their custom watcher, can follow the same rule.